### DISCUSSION TITLE
"take over" existing helm released application

### DISCUSSION THREAD

**Discussion by daveoy** [fluxcd/flux2#2848] [Q&A] [answered]
hi there,

im trying to use fluxcd to replace some old jenkins pipelines with quirky behaviour.

i have configured flux to release a helm chart that is already installed in the cluster with the same release name and namespace. flux appears to be succesffully reconciling the helmrelease, i can see the proper chart version being applied; however, the modified templates inside the chart do not appear to apply the changed resources. in addition to that, completely removing a resource that exists in the chart and reconciling the release doesn't appear to put the resources back.

flux tree ks flux-system shows the resources i deleted as a test from the helmchart so i believe flux thinks it has put that resource back. but it does not exist in the cluster

am i taking crazy pills? summoning @stefanprodan  the master of flux behaviour debugging

**Comment by daveoy** [comment, marked as answer]
as a test I modified the helmrelease to install into the namespace where the resources already live, rather than flux-system, and it appears to have not only replaced the resource i deleted, but also all the resources from that chart have been replaced. not sure if the resources from the original deployment were purged and written anew or if their creation timestamps were just overwritten.

**Reply by daveoy** [reply]
it looks like the resource removal issue is down to t he 'createNamespace: true' value on the helmrelease.

looks like i have my answer, thank you all.

**Reply by nalum** [reply]
Timely, this has just come up for me too. So with `createNamespace: false` you were able to have a Flux `HelmRelease` adopt an existing helm deployment?

**Reply by daveoy** [reply]
To be clear, what enabled flux to control an already installed chart for me was:

- The HelmRelease object needs to be created in the same namespace as the currently installed release

Some secondary behaviour I encountered:

- if you have createNamespace: true, you may run the risk of having the entire charts' templates that are installed into that namespace deleted, then recreated

I didn't dig too far into that behaviour, but it could be a mix of purge: true on the kustomization that applies the releases and createNamespace: true that created that condition for me yesterday.

That may not be a showstopper for you. It's just something to consider. You may want to test this behaviour more thoroughly than I have.

**Reply by nalum** [reply]
Thanks for the info @daveoy :+1: I'll dig a bit further and do some testing.

**Reply by daveoy** [reply]
quick update:

im noticing that flux / helm are now adding a string("+1") to the end of the chart version which is breaking  helm templates which put the chart version in metadata since "+" is an illegal character in those fields.

@Nalum / @stefanprodan might that be anything you've come across before?
